src/components/plots/FeaturesVsUMIsScatterplot.jsx

```diff
@@ -3,23 +3,29 @@ import PropTypes from 'prop-types';
 import { Vega } from 'react-vega';
 
 import EmptyPlot from './helpers/EmptyPlot';
+import transformOldFeaturesVsUMIsPlotData from './helpers/transformOldFeaturesVsUMIsPlotData';
 import generateSpec from '../../utils/plotSpecs/generateFeaturesVsUMIsScatterplot';
 
 const FeaturesVsUMIsScatterplot = (props) => {
   const {
     config, plotData, actions,
   } = props;
 
+  // can remove if we formally migrate
+  const newPlotData = (!Array.isArray(plotData) || !plotData.length)
+    ? plotData
+    : transformOldFeaturesVsUMIsPlotData(plotData);
+
   const [plotSpec, setPlotSpec] = useState(config);
 
   useEffect(() => {
-    if (config && plotData?.length) {
-      setPlotSpec(generateSpec(config, plotData));
+    if (config && newPlotData?.pointsData?.length) {
+      setPlotSpec(generateSpec(config, newPlotData));
     }
   }, [config, plotData]);
 
   const render = () => {
```

Comment:
instead of having:

```
return (
    <>
      { render()}
    </>
  );
```

Why don't you directly return whatever you need like that:

```
  if (!newPlotData?.pointsData?.length) {
    return (
      <EmptyPlot mini={config.miniPlot} />
    );
  }

  return (
    <center data-testid='vega-container'>
      <Vega spec={plotSpec} renderer='canvas' actions={actions} />
    </center>
  );
```
Seems much easier to read to me.

Reply:
Looks better to me too - great suggestion!